src/libappimage/desktop_integration.c

```diff
@@ -46,7 +46,12 @@ void traverse_handler_extract_relevant_desktop_integration_files(void* raw_handl
 
         create_parent_dir(target);
 
-        handler->extract_file(handler, entry_data, target);
+        char* link = handler->get_file_link(handler, entry_data);
+        if (link) {
+            appimage_extract_file_following_symlinks(handler->path, file_name, target);
+            free(link);
+        } else
```

Comment:
Please use braces on the else clause, too (makes code easier to read and prevents accidental control flow issues).

Reply:
Done.